pkg/aks/create.go

```diff
@@ -38,219 +38,106 @@ func CreateCluster(ctx context.Context, cred *Credentials, clusterClient *contai
-	}
-
-	// Network profile
-	if managedCluster.NetworkProfile != nil {
-		np := managedCluster.NetworkProfile
-		if np.NetworkPlugin != "" {
-			aksCluster.NetworkProfile.NetworkPlugin = np.NetworkPlugin
-		}
-		if np.NetworkPolicy != "" {
-			aksCluster.NetworkProfile.NetworkPolicy = np.NetworkPolicy
-		}
-		if np.NetworkMode != "" {
-			aksCluster.NetworkProfile.NetworkMode = np.NetworkMode
-		}
-		if np.DNSServiceIP != nil {
-			aksCluster.NetworkProfile.DNSServiceIP = np.DNSServiceIP
-		}
-		if np.DockerBridgeCidr != nil {
-			aksCluster.NetworkProfile.DockerBridgeCidr = np.DockerBridgeCidr
-		}
-		if np.PodCidr != nil {
-			aksCluster.NetworkProfile.PodCidr = np.PodCidr
-		}
-		if np.ServiceCidr != nil {
-			aksCluster.NetworkProfile.ServiceCidr = np.ServiceCidr
-		}
-		if np.OutboundType != "" {
-			aksCluster.NetworkProfile.OutboundType = np.OutboundType
-		}
-		if np.LoadBalancerSku != "" {
-			aksCluster.NetworkProfile.LoadBalancerSku = np.LoadBalancerSku
-		}
-		// LoadBalancerProfile is not configurable in Rancher so there won't be subfield conflicts. Just pull it from
-		// the state in AKS.
-	}
-
-	// Service principal client id and secret
-	if managedCluster.ServicePrincipalProfile != nil { // operator phase is 'updating' or 'active'
-		aksCluster.ServicePrincipalProfile = managedCluster.ServicePrincipalProfile
-	}
-
-	// Tags
-	if managedCluster.Tags != nil {
-		aksCluster.Tags = managedCluster.Tags
+// newManagedCluster creates a new managed Kubernetes cluster.
+func newManagedCluster(ctx context.Context, cred *Credentials, workplacesClient services.WorkplacesClientInterface, spec *aksv1.AKSClusterConfigSpec, phase string) (*containerservice.ManagedCluster, error) {
+	managedCluster := &containerservice.ManagedCluster{
+		Name:     to.StringPtr(spec.ClusterName),
+		Location: to.StringPtr(spec.ResourceLocation),
+		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
+			KubernetesVersion: spec.KubernetesVersion,
+		},
 	}
 
-	_, err = clusterClient.CreateOrUpdate(
-		ctx,
-		spec.ResourceGroup,
-		spec.ClusterName,
-		aksCluster,
-	)
-
-	return err
-}
-
-// CreateManagedCluster creates a new managed Kubernetes cluster.
-func CreateManagedCluster(ctx context.Context, cred *Credentials, spec *aksv1.AKSClusterConfigSpec, phase string) (containerservice.ManagedCluster, error) {
-	managedCluster := containerservice.ManagedCluster{}
-
-	// Get tags from config spec
 	tags := make(map[string]*string)
 	for key, val := range spec.Tags {
 		if val != "" {
 			tags[key] = to.StringPtr(val)
 		}
 	}
+	managedCluster.Tags = tags
 
-	// Get network profile from config spec
-	var vmNetSubnetID *string
 	networkProfile := &containerservice.NetworkProfile{
-		NetworkPlugin:   containerservice.Kubenet,
-		NetworkPolicy:   containerservice.NetworkPolicy(to.String(spec.NetworkPolicy)),
-		LoadBalancerSku: containerservice.Standard,
+		NetworkPolicy: containerservice.NetworkPolicy(to.String(spec.NetworkPolicy)),
```

Comment:
About `NetworkPolicy` in Azure there is limitation that `containerservice.NetworkPolicyCalico` can be used only with `containerservice.Kubenet`. When `containerservice.Kubenet` is used then NetworkPolicy can be `None` or `containerservice.NetworkPolicyCalico`

Reply:
For `NetworkPlugin == containerservice.Azure` NetworkPolicy can be `None` or `containerservice.NetworkPolicyCalico` or `containerservice.NetworkPolicyAzure`